LibreNMS/IPMI/IPMIClient.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * IPMIClient.php
+ *
+ * IPMI Client
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ * @copyright  2021 Trym Lund Flogard
+ * @author     Trym Lund Flogard <trym@flogard.no>
+ */
+
+namespace LibreNMS\IPMI;
+
+/**
+ * Represents an IPMI connection with a host machine.
+ */
+class IPMIClient
+{
+    private $ipmiToolPath;
+    private $host;
+    private $user;
+    private $password;
+    private $port = null;
+
+    private $privLvl = 'USER';
+    private $interface = 'lanplus';
+
+    /**
+     * Creates a new instance of the IPMIClient class.
+     * @param string $ipmiToolPath The absolute path to ipmitool.
+     * @param string $host The hostname or IP of the endpoint. Set to 'localhost' to connect via in-band driver.
+     * @param string $user
+     * @param string $password
+     */
+    public function __construct(string $ipmiToolPath, string $host, string $user, string $password)
+    {
+        $this->ipmiToolPath = $ipmiToolPath;
+        $this->host = $host;
+        $this->user = $user;
+        $this->password = $password;
+    }
+
+    public function getHost(): string
+    {
+        return $this->host;
+    }
+
+    /**
+     * Gets the IPMI interface driver used by the client.
+     */
+    public function getDriver(): string
+    {
+        return $this->interface;
+    }
+
+    /**
+     * Sets the IPMI interface driver.
+     */
+    public function setDriver(string $interface)
+    {
+        $this->interface = $interface;
+    }
+
+    /**
+     * Gets the port used by the client.
+     * @return ?string The port currently used. Null if not specified.
+     */
+    public function getPort(): ?string
+    {
+        return $this->port;
+    }
+
+    /**
+     * Set the port used by the client.
+     */
+    public function setPort(?string $port)
+    {
+        $this->port = $port;
+    }
+
+    /**
+     * Gets a binary representation of the cached SDR record for this host.
+     * @return string|false The SDR binary or false on failure.
+     */
+    public function getSDR()
+    {
+        $path = '/tmp/librenms/ipmi/SDR';
```

Comment:
Should be in the cache dir in librenms root instead

Reply:
Resolved in 88a8efd9834757c891497feb2de18f09aecd9a15.